script.js

```diff
@@ -1,4 +1,479 @@
+// set the different game turns
+var GAME_MODE_USERNAME = "USERNAME";
+var GAME_MODE_BETTING = "BETTING";
+var GAME_MODE_MULTI_PLAYER = "MULTI_PLAYER";
+var GAME_MODE_SPLITS = "SPLITS";
+var GAME_MODE_DEAL_CARDS = "DEAL_CARDS";
+var GAME_MODE_CHOOSE_ACE_VALUE = "CHOOSE_ACE_VALUE";
+var GAME_MODE_ANALYSE_CARDS = "ANALYSE_CARDS";
+var GAME_MODE_HIT_OR_STAND = "HIT_OR_STAND";
+
+// set the first game mode to input username
+var gameMode = GAME_MODE_USERNAME;
+
+// set different variables and arrays to hold values
+var userInput = "";
+var playerCards = [];
+var computerCards = [];
+var playerCard1;
+var playerCard2;
+var playerCard;
+var playerCardIndex = 0;
+var computerCard1;
+var computerCard2;
+var computerCard;
+var computerCardIndex = 0;
+var playerCardsTotal = 0;
+var computerCardsTotal = 0;
+var numOfPlayerCards = 0;
+var numOfComputerCards = 0;
+var playerBankroll = 100;
+var computerBankroll = 100;
+var betForRound = 0;
+var computerHasAce = false;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  userInput = input;
+  if (gameMode == GAME_MODE_USERNAME) {
+    myOutputValue = inputUsername();
+    return myOutputValue;
+  } else if (gameMode == GAME_MODE_BETTING) {
+    myOutputValue = inputBet();
+    return myOutputValue;
+  } else if (gameMode == GAME_MODE_DEAL_CARDS) {
+    myOutputValue = dealCards();
+  } else if (gameMode == GAME_MODE_CHOOSE_ACE_VALUE) {
+    myOutputValue = chooseAceValue();
+  } else if (gameMode == GAME_MODE_ANALYSE_CARDS) {
+    myOutputValue = analyseCards();
+  } else if (gameMode == GAME_MODE_HIT_OR_STAND) {
+    myOutputValue = hitOrStand();
+    return myOutputValue;
+  }
```

Comment:
I know this is not covered in our course material but you could consider using a switch statement here so that your code will look neater. The link to an explanation of a switch statement is https://www.w3schools.com/js/js_switch.asp